gamefunc.cpp

```diff
@@ -0,0 +1,925 @@
+    {
+        cout<<"Bledny tryb"<<endl;
+
+    }
+
+}
+
+void printBoard(figura board[8][8],int tryb /*1==biale 0==czarne*/)
+{
+    int i=1;
+#ifndef __unix__
+    HANDLE  hConsole;
+    hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+#endif
+
+#ifdef __unix__
+    // cout<<"\033[50:10x20";
+#endif // WIN32
+    if(tryb==1)
+        printf("  a b c d e f g h\n");
+    else
+        printf("  h g f e d c b a\n");
+
+    for(int y=0; y<8; y++)
+    {
+        i--;
+        printf("%d|",tryb==1 ? 8-(y) : y+1);
+        for(int x=0; x<8; x++)
+        {
+            if(i%2==1)
+            {
+#ifndef __unix__
+                if(board[tryb==1?7-y:y][tryb==0?7-x:x].tim==1)
+                    SetConsoleTextAttribute(hConsole, FOREGROUND_GREEN | FOREGROUND_GREEN | BACKGROUND_RED | BACKGROUND_GREEN | FOREGROUND_INTENSITY | BACKGROUND_BLUE | BACKGROUND_INTENSITY);
+                else if(board[tryb==1?7-y:y][tryb==0?7-x:x].tim==2)
+                    SetConsoleTextAttribute(hConsole, FOREGROUND_RED | BACKGROUND_RED | BACKGROUND_GREEN | FOREGROUND_INTENSITY | BACKGROUND_BLUE | BACKGROUND_INTENSITY);
+                else
+                    SetConsoleTextAttribute(hConsole, BACKGROUND_RED | BACKGROUND_GREEN | FOREGROUND_INTENSITY | BACKGROUND_BLUE | BACKGROUND_INTENSITY);
+#endif
+#ifdef __unix__
+
+                if(board[tryb==1?7-y:y][tryb==0?7-x:x].tim==1)
+                    cout<<"\u001b[37m\u001b[41m";//white
+                else if(board[tryb==1?7-y:y][tryb==0?7-x:x].tim==2)
+                    cout<<"\u001b[30m\u001b[41m";//black
+                else
+                    cout<<"\u001b[41m";
+#endif // WIN32
+
+            }
+            else
+            {
+#ifdef WIN32
+                if(board[tryb==1?7-y:y][tryb==0?7-x:x].tim==1)
+                    SetConsoleTextAttribute(hConsole, FOREGROUND_GREEN | FOREGROUND_GREEN| BACKGROUND_BLUE | FOREGROUND_INTENSITY | BACKGROUND_INTENSITY);
+                else if(board[tryb==1?7-y:y][tryb==0?7-x:x].tim==2)
+                    SetConsoleTextAttribute(hConsole, FOREGROUND_RED | BACKGROUND_BLUE | FOREGROUND_INTENSITY | BACKGROUND_INTENSITY);
+                else
+                    SetConsoleTextAttribute(hConsole, BACKGROUND_BLUE | FOREGROUND_INTENSITY | BACKGROUND_INTENSITY);
+#endif
+#ifndef WIN32
+                if(board[tryb==1?7-y:y][tryb==0?7-x:x].tim==1)
+                    cout<<"\u001b[37m\u001b[45m";//white
+                else if(board[tryb==1?7-y:y][tryb==0?7-x:x].tim==2)
+                    cout<<"\u001b[30m\u001b[45m";//black
+                else
+                    cout<<"\u001b[45m";
+#endif // WIN32
+
+            }
+            i++;
+
+            printf("%c ",board[tryb==1?7-y:y][tryb==0?7-x:x].fig);
+        }
+#ifdef WIN32
+        SetConsoleTextAttribute(hConsole, 0x7);
+#endif // WIN32
+#ifndef WIN32
+        cout<<"\u001b[0m";
+
+#endif // WIN32
+
+
+        printf("\n");
+    }
+}
+
+int isLegal(figura board[8][8],int yP,int xP,int yK,int xK,int cas[4])

```

Comment:
w C++ jest `bool`, który jest albo fałszem albo prawdą, nie trzeba się męczyć z "czy 0 to prawda czy fałsz?"